### ISSUE TITLE
🐛 Delta doesnt' show the entire diff

### ISSUE THREAD

**Issue by CGamesPlay** [dandavison/delta#1780]
```
$ delta --version
delta 0.17.0
$ cat diff.patch  | delta
```

Delta doesn't show the full patch. It stops after line 972 "Install the project...". The "---" lines look suspicious, but such lines appear elsewhere in the file (search "Enabling feature profiling") and work just fine. Snippet of where delta stops showing output:

```patch
-[ 87%] No install step for 'wasmtime-crate'
-[100%] Completed 'wasmtime-crate'
-[100%] Built target wasmtime-crate
-Install the project...
--- Install configuration: "Release"
--- Installing: /home/runner/work/wasmtime/wasmtime/target/c-api-install/include
--- Installing: /home/runner/work/wasmtime/wasmtime/target/c-api-install/include/wasm.h
--- Installing: /home/runner/work/wasmtime/wasmtime/target/c-api-install/include/wasmtime
--- Installing: /home/runner/work/wasmtime/wasmtime/target/c-api-install/include/wasmtime/async.h
--- Installing: /home/runner/work/wasmtime/wasmtime/target/c-api-install/include/wasmtime/profiling.h
--- Installing: /home/runner/work/wasmtime/wasmtime/target/c-api-install/include/wasmtime/global.h
```

[diff.patch](https://github.com/user-attachments/files/16463916/diff.patch)

**Comment by sskras** [comment]
I wonder if that has to do something with ANSI escape sequences occuring after the initial `-` a few lines below. Eg:
```diff
  ...
--- Installing: /home/runner/work/wasmtime/wasmtime/target/c-api-install/../c-api-install/lib/libwasmtime.so
--- Installing: /home/runner/work/wasmtime/wasmtime/target/c-api-install/../c-api-install/lib/libwasmtime.a
-##[group]Run ./ci/build-tarballs.sh "x86_64-android" "x86_64-linux-android"
-[36;1m./ci/build-tarballs.sh "x86_64-android" "x86_64-linux-android"[0m
  ...
```

**Comment by th1000s** [comment]
It's much simpler, `diff -u ..` input starts with (in this case) `--- /var/folder/...` to describe the original filenames, but removing a line starting with `--` (in this case `-- Enabling feature profiling`) creates a line that delta handles like a filename which results in garbled output.

Fixed in #1787
